application/src/main/java/org/opentripplanner/routing/algorithm/raptoradapter/transit/request/RouteRequestTransitDataProviderFilter.java

```diff
@@ -90,17 +87,18 @@ public static BikeAccess bikeAccessForTrip(Trip trip) {
   }
 
   @Override
-  public boolean tripPatternPredicate(TripPatternForDate tripPatternForDate) {
+  @Nullable
+  public Predicate<TripTimes> getTripFilter(TripPattern tripPattern) {
     for (TransitFilter filter : filters) {
-      if (filter.matchTripPattern(tripPatternForDate.getTripPattern().getPattern())) {
-        return true;
+      if (filter.matchTripPattern(tripPattern)) {
+        var withFilters = hasSubModeFilters && tripPattern.getContainsMultipleModes();
```

Comment:
`withFilters` should probably have a better name. What about `applyTripTimesFilters` ? This apply to the `tripTimesPredicate` method argument as well.